src/EOS/Address.cpp

```diff
@@ -0,0 +1,152 @@
+#include "Address.h"
+
+#include <stdexcept>
+#include <TrezorCrypto/base58.h>
+#include <TrezorCrypto/ripemd160.h>
+#include <BinaryCoding.h>
+
+using namespace TW::EOS;
+
+bool Address::isValid(const std::string& string) {
+    return extractKeyData(string);
+}
+
+/// Determines whether the given byte array is a valid keyBuffer
+/// Verifies the buffer's size and it's checksum bytes
+bool Address::isValid(const uint8_t *buffer, size_t size, EOS::Type type) {
+    if (size != KeyDataSize) return false;
+
+    // last Address::ChecksumSize bytes are a checksum
+    uint32_t checksum = decode32LE(buffer + PublicKeyDataSize);
+    if (createChecksum(buffer, type) != checksum) return false;
+    return true;
+}
+
+/// Creates a checksum of PublicKeyDataSize bytes at the buffer
+/// IMPORTANT: THERE ARE NO SIZE CHECKS. THE BUFFER IS ASSUMED
+///             TO HAVE AT LEAST PublicKeyDataSize bytes.
+uint32_t Address::createChecksum(const uint8_t *buffer, Type type) {
+    // create our own checksum and compare the two
+    uint8_t hash[RIPEMD160_DIGEST_LENGTH];
+    RIPEMD160_CTX ctx;
+    ripemd160_Init(&ctx);
+
+    // add the buffer bytes to the hash input
+    ripemd160_Update(&ctx, buffer, PublicKeyDataSize);
+
+    //  append the prefix to the hash input as well in case of modern types
+    switch (type) {
+    case Type::Legacy: // no extra input
+        break;
+
+    case Type::ModernK1:
+        ripemd160_Update(&ctx, 
+                        (const uint8_t *) Modern::K1::prefix.c_str(), 
+                        static_cast<uint32_t>(Modern::K1::prefix.size()));
+        break;
+
+    case Type::ModernR1:
+        ripemd160_Update(&ctx, 
+                        (const uint8_t *) Modern::R1::prefix.c_str(), 
+                        static_cast<uint32_t>(Modern::R1::prefix.size()));
+        break;
+    }
+
+    // finalize the hash
+    ripemd160_Final(&ctx, hash);
+
+    return decode32LE(hash);
+}
+
+/// Extracts and verifies the key data from a base58 string.
+/// If the second arg is provided, the keyData and isTestNet 
+/// properties of that object are set from the extracted data.
+bool Address::extractKeyData(const std::string& string, Address *address) {
+    // verify if the string has one of the valid prefixes
+    Type type;
+    size_t prefixSize;
+    // std::cout << "Checking prefix for " + string + ".\n";
```

Comment:
let's remove this test code and legacy support

Reply:
okay